### DISCUSSION TITLE
Callbacks as event

### DISCUSSION THREAD

**Discussion by timhoeppner** [tailwindlabs/headlessui#340] [General]
I'm just curious, why are the callback functions on the components implemented as properties instead of events? I'm looking at the Vue implementation.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

Which components are you referring to exactly?

**Reply by timhoeppner** [reply]
Modal -> onClose
Transition -> beforeEnter, afterEnter, beforeLeave, afterLeave
